### PR TITLE
Fix UACC paths in Close()

### LINKED ISSUE #17769: `failed to open user account database` on session close
**Expected behavior:**

There is no error error upon exiting a terminal

**Current behavior:**

```
chris@pioneer ~ » tsh ssh root@cherry
root@cherry ~ # exit
Failed to launch: failed to open user account database, code: %!d(<nil>).
ERROR: Process exited with status 255
chris@pioneer ~ » 
```

**Bug details:**
- Teleport version `Teleport v11.0.0 git:v11.0.0-0-g0c2973d go1.19.2`
- Recreation steps: 
  1. connect to node
  2. end the session
  3. see the error message

<details>

<summary>Click to expand the debug logs</summary>

```
INFO [CLIENT]    [KEY AGENT] Connected to the system agent: "/tmp/ssh-XXXXXXufhPDF/agent.2569" client/api.go:4236
DEBU [KEYSTORE]  Returning Teleport TLS certificate "/home/chris/.tsh/keys/teleport.cluster.example/christian-heusel-x509.pem" valid until "2022-10-25 10:34:50 +0000 UTC". client/keystore.go:325
DEBU [KEYSTORE]  Reading certificates from path "/home/chris/.tsh/keys/teleport.cluster.example/christian-heusel-ssh/teleport.cluster.example-cert.pub". client/keystore.go:348
INFO [KEYAGENT]  Loading SSH key for user "christian-heusel" and cluster "teleport.cluster.example". client/keyagent.go:248
DEBU [KEYSTORE]  Returning Teleport TLS certificate "/home/chris/.tsh/keys/teleport.cluster.example/christian-heusel-x509.pem" valid until "2022-10-25 10:34:50 +0000 UTC". client/keystore.go:325
INFO [CLIENT]    Connecting to proxy=teleport.cluster.example:443 login="christian-heusel" using TLS Routing client/api.go:3187
DEBU [KEYSTORE]  Returning Teleport TLS certificate "/home/chris/.tsh/keys/teleport.cluster.example/christian-heusel-x509.pem" valid until "2022-10-25 10:34:50 +0000 UTC". client/keystore.go:325
DEBU [HTTP:PROX] No proxy set in environment, returning direct dialer. proxy/proxy.go:301
DEBU [KEYSTORE]  Returning Teleport TLS certificate "/home/chris/.tsh/keys/teleport.cluster.example/christian-heusel-x509.pem" valid until "2022-10-25 10:34:50 +0000 UTC". client/keystore.go:325
DEBU [KEYAGENT]  "Checking key: ssh-rsa-cert-v01@openssh.com AAAAHHNzaC1yc2EtY2VydC12MDFAb3BlbnNzaC5jb20AAAAgTAid4csjJZN3zPV6QuKYlnm0RNNR2ppEk0JD6AnU2tMAAAADAQABAAABAQDsJHSraL4fcZiPOOwgq+idCBhDOuXdwftm8PXIKyaPotaHbStV02a/GSDavq8gSNipBvBL1I7qI3V9gaoAeiEAItEU+AGwTLDWbrW6XSYjUWAU6I8JDJQHuBO2xrFsyX7rvxOeIOKE0prCA4zl6nX1rsJwQL+TzkeY7jMKy+RMuwkFaVCqy+GOy81Je6bTxszrTicTxLL7NdDs8FacdgvdYWiDsFzMxZ4SEz9TYa49qe1EUXDX4LVxqtPlWY1sSGsNrb0ScxnhKAtLR591dN3qeHBdLsDfzov8mbnlP0Eor1bKMbbLdmWrXaXBru5JYQb+tQTmKf9oolxqxcIUMGp5AAAAAAAAAAAAAAACAAAAAAAAAPcAAAA7MzQ3NjZlZTMtNmRiNi00ZmI3LWJhZjMtZWY5MjEyNDUxMTVmLnRlbGVwb3J0Lm1hdGhwaHlzLmluZm8AAAAkMzQ3NjZlZTMtNmRiNi00ZmI3LWJhZjMtZWY5MjEyNDUxMTVmAAAAHGFwcGxlLnRlbGVwb3J0Lm1hdGhwaHlzLmluZm8AAAAFYXBwbGUAAAAJbG9jYWxob3N0AAAACTEyNy4wLjAuMQAAAAM6OjEAAAAWdGVsZXBvcnQubWF0aHBoeXMuaW5mbwAAAChyZW1vdGUua3ViZS5wcm94eS50ZWxlcG9ydC5jbHVzdGVyLmxvY2FsAAAAAGLsBtD//////////wAAAAAAAABWAAAAFHgtdGVsZXBvcnQtYXV0aG9yaXR5AAAAGgAAABZ0ZWxlcG9ydC5tYXRocGh5cy5pbmZvAAAAD3gtdGVsZXBvcnQtcm9sZQAAAAkAAAAFUHJveHkAAAAAAAABFwAAAAdzc2gtcnNhAAAAAwEAAQAAAQEAyPCk/cefkkoTKxq/zUoRdv32dmpF5YDSWuKUwWPUrSS6rZScyqm2AlU6FB3eRk6xr+5S5scuK4wjBWo4b7Vb+nwjSzDno3g3aC4518OtTJGt4FpjkE59JZIJU8a9X8TDSRV4uLZg5QRxQ6cSITFTat9EE9WMQ6cUhNBjURKTKaeRXS1SbcrsLQ04ClXdltHWefSSA4aOZ8Jjn1I4AvhJpQKl4/AdYME5hDt0+Qv8TtVp6KpI9CPdxgb3WZkj3ZqD2+09w9l4p0Jlj9IMuM5oZY0hYopkVCgszrUY2T1W4W2ILPezhWjfkFq8TOzrZRsVTWkTOE/E2NwYJX2qWoGuCwAAARQAAAAMcnNhLXNoYTItNTEyAAABAMQ1bXJJiEPFD+XUqOTUWPaTAV4tKdtDfYfKYhKbGaNMrFunAWavLDp0CJsYVpxwefnqon+Hj4wljwj3w2fc4OpLQ+81xLGtmF41nz9KuqZwIR7EdkaUcAeW/MWozab+ukraPoFTWB+ZgW/p2T6Eqg+PK+AAbUiw4t1g3/SmgLXdI/ieYe2Ub4gLOwP1mDQOTdKPuHFtaRSFRgAtSt6f6Yl2NqTPtjBLferszlTbmgERDOqqGKx21B8lTUdEIQVywUk++PJm8tr13yEpDas5XHmufXElN9teeJwupG3Q2FyiAFHzoVnTqn5L7PIHv5XnPBWtgb1QgpXdXjj173iUojY=\n." client/keyagent.go:420
DEBU [KEYAGENT]  Validated host teleport.cluster.example:443. client/keyagent.go:426
INFO [CLIENT]    Successful auth with proxy teleport.cluster.example:443. client/api.go:3192
DEBU [KEYSTORE]  Returning Teleport TLS certificate "/home/chris/.tsh/keys/teleport.cluster.example/christian-heusel-x509.pem" valid until "2022-10-25 10:34:50 +0000 UTC". client/keystore.go:325
INFO [CLIENT]    Client= connecting to node=cherry on cluster teleport.cluster.example client/client.go:1514
DEBU [KEYSTORE]  Returning Teleport TLS certificate "/home/chris/.tsh/keys/teleport.cluster.example/christian-heusel-x509.pem" valid until "2022-10-25 10:34:50 +0000 UTC". client/keystore.go:325
DEBU [KEYSTORE]  Returning Teleport TLS certificate "/home/chris/.tsh/keys/teleport.cluster.example/christian-heusel-x509.pem" valid until "2022-10-25 10:34:50 +0000 UTC". client/keystore.go:325
DEBU [KEYSTORE]  Reading certificates from path "/home/chris/.tsh/keys/teleport.cluster.example/christian-heusel-ssh/teleport.cluster.example-cert.pub". client/keystore.go:348
DEBU [CLIENT]    MFA not required for access. client/client.go:439
DEBU [KEYSTORE]  Returning Teleport TLS certificate "/home/chris/.tsh/keys/teleport.cluster.example/christian-heusel-x509.pem" valid until "2022-10-25 10:34:50 +0000 UTC". client/keystore.go:325
DEBU [KEYAGENT]  "Checking key: ssh-rsa-cert-v01@openssh.com AAAAHHNzaC1yc2EtY2VydC12MDFAb3BlbnNzaC5jb20AAAAgtQ/ZnKUSmK4VdR55jWgB/a46ea2CWLxiyKDC+PGk+aEAAAADAQABAAABAQDLLfEwwM6pOh5UCGTvBsUFLnbx9JcO9iS6pwEQGnJBFRWI+s0im7T+qEDNByJTVQRLiXgA4UjOowSQWXWtROh4XoCOFVDW+beR/FUIsFO10F3CFNUK8mobrnOPs73phsk3BJ6mKz+xPybsK1VsFeQBsBgKLibgiI2Nx8fHBlp/uhVlDyiqyEc2vvgDXwxHnmHoDczvCosRf6FaX/VVYGAatcivdRN+pBxOc1jhlelkfk7JZZqhuDniVW+tJUX/HEWgAjyOIIPMICZNRqNShu8AJEPKN4n616PaO8dxWhKMwdY7GCodQ93WDjn1r2Pc04ha5EDjmcdNSMDEMGOPx0R5AAAAAAAAAAAAAAACAAAAAAAAAMUAAAA7OTBhYWU0MTAtYjFlNS00NDY3LWIzZjQtNTQ1NmZiNGIwMDczLnRlbGVwb3J0Lm1hdGhwaHlzLmluZm8AAAAkOTBhYWU0MTAtYjFlNS00NDY3LWIzZjQtNTQ1NmZiNGIwMDczAAAAHWNoZXJyeS50ZWxlcG9ydC5tYXRocGh5cy5pbmZvAAAABmNoZXJyeQAAAAlsb2NhbGhvc3QAAAAJMTI3LjAuMC4xAAAAAzo6MQAAAA4xMjkuMjA2LjI3LjI0MgAAAABi7BUt//////////8AAAAAAAAAVQAAABR4LXRlbGVwb3J0LWF1dGhvcml0eQAAABoAAAAWdGVsZXBvcnQubWF0aHBoeXMuaW5mbwAAAA94LXRlbGVwb3J0LXJvbGUAAAAIAAAABE5vZGUAAAAAAAABFwAAAAdzc2gtcnNhAAAAAwEAAQAAAQEAyPCk/cefkkoTKxq/zUoRdv32dmpF5YDSWuKUwWPUrSS6rZScyqm2AlU6FB3eRk6xr+5S5scuK4wjBWo4b7Vb+nwjSzDno3g3aC4518OtTJGt4FpjkE59JZIJU8a9X8TDSRV4uLZg5QRxQ6cSITFTat9EE9WMQ6cUhNBjURKTKaeRXS1SbcrsLQ04ClXdltHWefSSA4aOZ8Jjn1I4AvhJpQKl4/AdYME5hDt0+Qv8TtVp6KpI9CPdxgb3WZkj3ZqD2+09w9l4p0Jlj9IMuM5oZY0hYopkVCgszrUY2T1W4W2ILPezhWjfkFq8TOzrZRsVTWkTOE/E2NwYJX2qWoGuCwAAARQAAAAMcnNhLXNoYTItNTEyAAABAA1W1zMxjJc0XtbFiA3XzxNGD9f4IMXLzkJfB7shHAf++jiotRlSsz2r8E/j+5cIEbWeqtwzZMwUBmy4rAaKXzwIud08schYhsJTeZfDZNgMn9fG0vIlJlD/VEZoNR28yXnYIUl+Qbl71rUk3v0qExgkkqmNOgC25Aj07tK3jjBMrmUryv1EmevNThkd4UdsC0AKHfH09PlAoydk1/jDpdAsxot7CReipiT527D0kWlQnsdkTbrqx2ve+7cCZkfmVYOZDoJltS7E3DGA7CYVjsG5HGDG3vmZ6mAGiNMjxnoWffKv3/xOlcekZWgZlIFuCcyqAFi9vti4mjIYMDlBTzo=\n." client/keyagent.go:420
DEBU [KEYAGENT]  Validated host cherry:0@default@teleport.cluster.example. client/keyagent.go:426
DEBU [CLIENT]    No Key Agent selected. client/session.go:279
root@cherry ~ # exit
Failed to launch: failed to open user account database, code: %!d(<nil>).

ERROR REPORT:
Original Error: *ssh.ExitError Process exited with status 255
Stack Trace:
	github.com/gravitational/teleport/lib/client/api.go:2980 github.com/gravitational/teleport/lib/client.(*TeleportClient).runShell
	github.com/gravitational/teleport/lib/client/api.go:1990 github.com/gravitational/teleport/lib/client.(*TeleportClient).runShellOrCommandOnSingleNode
	github.com/gravitational/teleport/lib/client/api.go:1934 github.com/gravitational/teleport/lib/client.(*TeleportClient).SSH
	github.com/gravitational/teleport/tool/tsh/tsh.go:2666 main.onSSH.func1.1
	github.com/gravitational/teleport/lib/client/api.go:731 github.com/gravitational/teleport/lib/client.RetryWithRelogin
	github.com/gravitational/teleport/tool/tsh/tsh.go:2665 main.onSSH.func1
	github.com/gravitational/teleport/tool/tsh/tsh.go:2578 main.retryWithAccessRequest
	github.com/gravitational/teleport/tool/tsh/tsh.go:2664 main.onSSH
	github.com/gravitational/teleport/tool/tsh/tsh.go:978 main.Run
	github.com/gravitational/teleport/tool/tsh/tsh.go:438 main.main
	runtime/proc.go:250 runtime.main
	runtime/asm_amd64.s:1594 runtime.goexit
User Message: Process exited with status 255
```
</details>

### PR DESCRIPTION
**Pull request by jakule**
Fixes https://github.com/gravitational/teleport/issues/17769

### CODE REVIEW

File: integration/utmp_integration_test.go [hunk 1]
```diff
@@ -143,6 +143,9 @@ func TestRootUsernameLimit(t *testing.T) {
 	username = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa"
 	err = uacc.Open(utmpPath, wtmpPath, username, "localhost", host, tty)
 	require.NoError(t, err)
+
+	err = uacc.Close(utmpPath, wtmpPath, tty)
```
Comment (codingllama) [inline integration/utmp_integration_test.go:147]: I see 2 opens above but only one close, is that OK?

Should we close in a defer or t.Cleanup to make sure it always happens?

Comment (jakule) [reply integration/utmp_integration_test.go:147]: I think it's ok.
1. The first Open is expected to fail, so we should not call close.
2. The second Open() should be closed, but only if there is no failure.

`defer` would need to be in the same place as `Close()` is right now (at the very end of the function). For me sounds like noise.

Comment (codingllama) [reply integration/utmp_integration_test.go:147]: 👍

**Review by codingllama — APPROVED** [review]

File: lib/srv/uacc/uacc_linux.go [hunk 2]
```diff
@@ -195,6 +182,24 @@ func Close(utmpPath, wtmpPath string, tty *os.File) error {
 	}
 }
 
+// getDefaultPaths sets the default paths for utmp and wtmp files if passed empty.
+func getDefaultPaths(utmpPath, wtmpPath string) (string, string) {
```
Comment (r0mant) [inline lib/srv/uacc/uacc_linux.go:186]: Nit: Use named return args for better clarity.
```suggestion
func getDefaultPaths(utmpPath, wtmpPath string) (utmpPath, wtmpPath string) {
```

File: lib/srv/uacc/uacc_linux.go [hunk 3]
```diff
@@ -195,6 +182,24 @@ func Close(utmpPath, wtmpPath string, tty *os.File) error {
...
+	if utmpPath == "" {
+		utmpPath = utmpFilePath
+	}
+
+	if wtmpPath == "" {
+		// Check where wtmp is located.
+		if utils.FileExists(wtmpFilePath) {
+			wtmpPath = wtmpFilePath
+		} else {
+			wtmpPath = wtmpAltFilePath
```
Comment (r0mant) [inline lib/srv/uacc/uacc_linux.go:196]: Does `wtmpAltFilePath` always exist? And won't we get the same issue if it doesn't?

Comment (jakule) [reply lib/srv/uacc/uacc_linux.go:196]: If it doesn't exist, this function should not be called. Out logic in reexec takes this into account. I can refactor the code to make it more explicit.

Comment (r0mant) [reply lib/srv/uacc/uacc_linux.go:196]: I think a comment would suffice in that case.

**Review by r0mant — APPROVED** [review]

File: lib/srv/uacc/uacc_linux.go (hunk 2 above)
Comment (jakule) [reply lib/srv/uacc/uacc_linux.go:186]: This is not going to work:
```
# github.com/gravitational/teleport/lib/srv/uacc
lib/srv/uacc/uacc_linux.go:186:50: utmpPath redeclared in this block
        lib/srv/uacc/uacc_linux.go:186:22: other declaration of utmpPath
lib/srv/uacc/uacc_linux.go:186:60: wtmpPath redeclared in this block
        lib/srv/uacc/uacc_linux.go:186:32: other declaration of wtmpPath
```
I can rename it, but at this point, this sounds a bit "forced".

Comment (r0mant) [reply lib/srv/uacc/uacc_linux.go:186]: Ah yeah, good call. Not a big deal.

**Pull request merged**